ethportal-peertest/src/lib.rs

```diff
@@ -2,31 +2,67 @@ pub mod cli;
 pub mod jsonrpc;
 
 pub use cli::PeertestConfig;
-pub use jsonrpc::{get_enode, test_jsonrpc_endpoints_over_http, test_jsonrpc_endpoints_over_ipc};
+pub use jsonrpc::get_enode;
 
 use std::sync::Arc;
+use std::{thread, time};
+
+use futures::future;
 
 use trin_core::cli::TrinConfig;
 use trin_core::jsonrpc::service::JsonRpcExiter;
+use trin_core::portalnet::types::messages::SszEnr;
+
+pub struct PeertestNode {
+    pub enr: SszEnr,
+    pub exiter: Arc<JsonRpcExiter>,
+}
 
-pub async fn launch_buddy_node() -> (String, Arc<JsonRpcExiter>) {
-    // Run a client, as a buddy peer for ping tests, etc.
-    let trin_config = TrinConfig::new_from(
-        [
+pub async fn launch_node(id: u16, bootnode_enr: Option<&String>) -> anyhow::Result<PeertestNode> {
+    let discovery_port: u16 = 9000 + id;
+    let discovery_port: String = discovery_port.to_string();
+    let web3_ipc_path = format!("/tmp/ethportal-peertest-buddy-{id}.ipc");
+    let trin_config_args: Vec<&str> = match bootnode_enr {
+        Some(enr) => vec![
             "trin",
             "--internal-ip",
-            "--unsafe-private-key",
-            "7261696e626f7773756e69636f726e737261696e626f7773756e69636f726e73",
```

Comment:
Ah, I see that this happened to support multiple concurrent peers. I thought we used the consistency of the enode in some test, so I'm just curious how that got resolved... Now I'm thinking that maybe there was a plan to test the enode of the test node and we never actually wrote that test.